plugins/core/hu.bme.mit.gamma.expression.model/src/hu/bme/mit/gamma/expression/util/ExpressionModelValidator.java

```diff
@@ -651,4 +668,75 @@ else if (variableDeclarationType instanceof TypeReference &&
 		return validationResultMessages;
 	}
 	
+//////////////////////////////////////////////////////////////////////
+	public Collection<ValidationResultMessage> checkArrayTypeDefinition(ArrayTypeDefinition arrayType){
+		Collection<ValidationResultMessage> validationResultMessages = new ArrayList<ValidationResultMessage>();
+		// Array init size must greater then 0
+		if(expressionEvaluator.evaluateInteger(arrayType.getSize()) <= 0) {
+			validationResultMessages.add(new ValidationResultMessage(ValidationResult.ERROR,
+					"The size of the array must be greater then 0.",
+					new ReferenceInfo(ExpressionModelPackage.Literals.ARRAY_TYPE_DEFINITION__SIZE, null)));
+		}
+		return validationResultMessages;
+	}	
+	
+	public Collection<ValidationResultMessage> checkSelfComparison(PredicateExpression expression){
+		Collection<ValidationResultMessage> validationResultMessages = new ArrayList<ValidationResultMessage>();		
+		// BinaryExpression
+		if(expression instanceof BinaryExpression) {
+			BinaryExpression BinaryExpression = (BinaryExpression) expression;
+			// the left and the right hand sides same
+			if(EcoreUtil.equals(BinaryExpression.getLeftOperand(), BinaryExpression.getRightOperand())) {
+				// EquivalenceExpression
+				if(expression instanceof EquivalenceExpression) {
+					EquivalenceExpression equivalenceExpression = (EquivalenceExpression) expression;
+					// EqualityExpression
+					if(equivalenceExpression instanceof EqualityExpression) {
+						validationResultMessages.add(new ValidationResultMessage(ValidationResult.ERROR,
```

Comment:
Why is this an error? The model remains correct even if it contains an expression like `true == true`. I think this should be an `INFO`.